services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsJsonObject.java

```diff
@@ -33,6 +36,41 @@ public String toJson() {
     return gson.create().toJson(this);
   }
 
+  /**
+   * Use this method to access JSON property that wasn't recognized during JSON serialization.
+   * This may be useful to access experimental properties.
+   * @param propertyName name of a json property
+   * @return value of a requested property or null if the requested property doesn't exist.
+   */
+  @Nullable
+  public final JsonElement getUnrecognizedProperty(String propertyName) {
```

Comment:
suppose I missed: why do we just not return the `map<String, JsonElement>`?

Reply:
Because under the hood they are `Map<String, SerializableJsonElement>`. From the other side, why not transform one map to another every time field is accessed? what do you think?

Reply:
We need to expose unrecognized properties to support experimental API features (like EV routing waypoint properties or state of charge annotations).